modules/ingester/instance.go

```diff
@@ -113,26 +114,32 @@ func (i *instance) PushBytes(ctx context.Context, id tempodb_encoding.ID, object
 
 // Moves any complete traces out of the map to complete traces
 func (i *instance) CutCompleteTraces(cutoff time.Duration, immediate bool) error {
+	now := time.Now()
+	tracesToCut := list.New()
+
 	i.tracesMtx.Lock()
-	defer i.tracesMtx.Unlock()
+	for key, trace := range i.traces {
```

Comment:
What about pulling this out into its own method that returns a tracesToCut slice, then you cam move the i.traceMtx locking into that method as well.

Reply:
Yeah, this could help with readability.  A defer always feels like a cleaner/safer way to unlock.